pim-tool-back/src/main/java/pilotproject/Project_Management_ELCA/repository/custom/ProjectRepositoryCustomImpl.java

```diff
@@ -0,0 +1,52 @@
+package pilotproject.Project_Management_ELCA.repository.custom;
+
+import com.querydsl.core.BooleanBuilder;
+import com.querydsl.jpa.impl.JPAQuery;
+import pilotproject.Project_Management_ELCA.model.entity.Project;
+import pilotproject.Project_Management_ELCA.model.entity.QProject;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.List;
+import java.util.Optional;
+
+public class ProjectRepositoryCustomImpl implements ProjectRepositoryCustom{
+    @PersistenceContext
+    private EntityManager em;
+
+    @Override
+    public Project findProjectByNumber(Integer number) {
+        return new JPAQuery<Project>(em).from(QProject.project)
+                .where(QProject.project.projectNumber.eq(number))
+                .fetchOne();
+    }
+
+    @Override
+    public List<Project> findProjectBySearchTextAndStatus(String searchText, String status) {
+        BooleanBuilder predicate = new BooleanBuilder();
+
+
+        if (!"null".equals(searchText)) {
```

Comment:
[S] What if the user want to search a project having name as `null`? Frontend should just pass null value to backend instead of a string.,